File: src/tools/scan_cli.cpp
//
// Created by wittgen on 3/29/22.
//
#include "ScanConsole.h"

int main(int argc, char *argv[]) {
    ScanConsole con;
    int res=con.init(argc, argv);
    if(res<=0) return res;
    res=con.loadConfig(); if(res!=0) exit(res);
    res=con.initHardware(); if(res!=0) exit(res);
    res=con.configure(); if(res!=0) exit(res);
    res=con.setupScan(); if(res!=0) exit(res);
    con.run();
    con.cleanup();
    con.plot();
    return 0;
}

File: src/python/tools/scan_cli.py
#!/usr/bin/env python3
import sys
import pyYARR


def run_scan():
    con = pyYARR.ScanConsole()
    res = con.init(sys.argv)
    if res <= 0:
        return res
    res = con.loadConfig()
    if res != 0:
        return res
    res = con.initHardware()
    if res != 0:
        return res
    res = con.configure()
    if res != 0:
        return res
    res = con.setupScan()
    if res != 0:
        return res
    con.run()
    con.cleanup()
    con.plot()
    return 0

if __name__ == '__main__':
    run_scan()
